client/components/Cart.js

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import axios from "axios";
+import { fetchCart, deleteFromCart } from "../store/cart";
+
+class Cart extends Component {
+  constructor(props) {
+    super(props);
+    //this.deleteCartItem = this.deleteCartItem.bind(this);
+  }
+
+  async componentDidMount() {
+    this.props.fetchCart();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.products !== prevProps.products) {
+      this.props.fetchCart();
+    }
+  }
+
+  // deleteCartItem(lineId, lineQuantity) {
+  //   this.props.deleteLineitem(lineId, lineQuantity);
+  // }
+
+  render() {
+    const { cart } = this.props;
+    console.log('cart', cart);
+    return (
+      <div>
+        <h2>Cart:</h2>
+        {cart.map((item) => {
+          return (
+            <div key={item.id}>
+              <ul>
+                {
+                item.lineitems.map((line) => {
+                  return (
+                    <li key={line.id}>
+                      {line.product?.name}({line.quantity})
+                      <button
+                        className="delete"
+                        type="delete"
+                        onClick={() =>
+                          this.props.deleteLineitem(line.id, line.quantity)
+                        }
+                      >
+                        {" "}
+                        Delete{" "}
+                      </button>
+                    </li>
+                  );
+                })}
+              </ul>
+            </div>
+          );
+        })}
+      </div>
+    );
+  }
+}
+
+const mapState = (state) => state;
```

Comment:
i see in the component, you're only using the `cart` state. since you're only using the cart state, you dont have to return the entire state of the store here
```suggestion
const mapState = ({ cart }) => ({ cart });
```